syntax = "proto3";

import "base_definitions.proto";

package FIRST;

//CreateOrderOperation
message CreateOrderRequest {

    uint32 request_id = 1;

    //uint64 in_timestamp = 2;

    OrderField order = 3;

    //uint64 out_timestamp = 4;
}

message CreateOrderResponse {

    uint32 request_id = 1;

    //uint64 in_timestamp = 2;

    OrderField order = 3;

    //uint64 out_timestamp = 4;

    enum Status {
        SUCCESS = 0;
        REJECTED = 1;
    }
    Status status = 5;
    string reason = 6;
}

//
message CancelOrderRequest {
    uint32 request_id = 1;

    //uint64 in_timestamp = 2;

    uint64 client_order_id = 3;

    OrderField order = 4;

    //uint64 out_timestamp = 5;
}

message CancelOrderResponse {
    uint32 request_id = 1;

    //uint64 in_timestamp = 2;

    OrderField order = 3;

    //uint64 out_timestamp = 4;
    enum Status {
        SUCCESS = 0;
        REJECTED = 1;
    }
    Status status = 5;
    string reason = 6;
}

//OrderFeed
message OrderFeed {
    uint64 request_id = 1;
    uint64 in_timestamp = 2;
    OrderField order = 3;
    uint64 out_timestamp = 4;
}

//TradeFeed
message TradeFeed {
    uint64 request_id = 1;
    uint64 in_timestamp = 2;
    TradeField  trade = 3;
    uint64 out_timestamp = 4;
}
